numpy/lib/shape_base.py

```diff
@@ -27,14 +27,24 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
     Execute `func1d(a, *args)` where `func1d` operates on 1-D arrays and `a`
     is a 1-D slice of `arr` along `axis`.
 
+    For each `i...`, `j...`, `k...` in turn (representing a series of indices),
+    where the number of indices in `i...` is equal to `axis`, this computes::
```

Comment:
Umm, clear as mud. Also, most uses of "( ... )" are better punctuated with commas.

Reply:
I don't understand your point about `...`

Reply:
Chuck is on a quest against parenthetical comments :)   (not ellispsis)

(rightly I think, replacing them with sentence-clauses/commas improved my doc commits in #9056)

Reply:
Maybe this would be better if you could write some short python pseudocode for what happens? Eg
```python
for i,j,k in np.ndindex(src.shape[0], len(indices), src.shape[2]):
    dst[i,j,k] = src[i,indices[j],k]
```

Reply:
```python
replace = lambda lst,i,v: lst[:i] + (v,) + lst[i+1:]

dst = np.empty(replace(src.shape, axis, len(indices)))
for dst_indx in np.ndindex(dst.shape):
    src_indx = replace(dst_indx, axis, indices[dst_indx[axis]])
    dst[dst_indx] = src[src_indx]
```

Reply:
How about:
```
for ii in np.ndindex(dst.shape[:axis]):
    for kk in np.ndindex(dst.shape[axis+1:]):
        out[ii + s_[...] + kk] = a[ii + s_[:] + kk][indices]
```

Using `s_` as a shorthand for `np.s_`

Reply:
hey, that's not too bad. I find it fairly legible.

Stepping back and squinting, I like the first one better, especially if you add the comment from the second.